src/plugins/telemetry/server/plugin.ts

```diff
@@ -42,19 +43,26 @@ import {
 import { TelemetryConfigType } from './config';
 import { FetcherTask } from './fetcher';
 import { handleOldSettings } from './handle_old_settings';
+import { getTelemetrySavedObject } from './telemetry_repository';
+import { getTelemetryOptIn } from '../common/telemetry_config';
 
-export interface TelemetryPluginsSetup {
+interface TelemetryPluginsDepsSetup {
   usageCollection: UsageCollectionSetup;
   telemetryCollectionManager: TelemetryCollectionManagerPluginSetup;
 }
 
-export interface TelemetryPluginsStart {
+interface TelemetryPluginsDepsStart {
   telemetryCollectionManager: TelemetryCollectionManagerPluginStart;
 }
 
+export type TelemetryPluginsSetup = void;
+export interface TelemetryPluginsStart {
+  getIsOptedIn: () => Promise<boolean | null>;
```

Comment:
Let's add some public documentation for our team-mates (and our future selves) about when/why it may return `null`. Personally, I would expect the method to always return a boolean: either from the SOs or from the config 😉